catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswSubscriptionEndpoint.java

```diff
@@ -52,45 +65,61 @@
 import org.osgi.service.cm.ConfigurationAdmin;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.w3c.dom.Node;
 
+import ddf.catalog.data.Metacard;
+import ddf.catalog.event.EventProcessor;
+import ddf.catalog.event.Subscriber;
 import ddf.catalog.event.Subscription;
 import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
 import net.opengis.cat.csw.v_2_0_2.AcknowledgementType;
 import net.opengis.cat.csw.v_2_0_2.EchoedRequestType;
+import net.opengis.cat.csw.v_2_0_2.GetRecordsResponseType;
 import net.opengis.cat.csw.v_2_0_2.GetRecordsType;
 import net.opengis.cat.csw.v_2_0_2.ObjectFactory;
 import net.opengis.cat.csw.v_2_0_2.QueryType;
 
-public class CswSubscriptionEndpoint implements Subscribe {
+@Path("/subscription")
+public class CswSubscriptionEndpoint implements CswSubscribe, Subscriber {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CswSubscriptionEndpoint.class);
 
-    private Map<String, ServiceRegistration<Subscription>> registeredSubscriptions =
-            new ConcurrentHashMap<>();
-
-    private ObjectFactory objectFactory = new ObjectFactory();
+    private static final String METACARD_SCHEMA = "urn:catalog:metacard";
 
     private static final String UUID_URN = "urn:uuid:";
 
-    private DatatypeFactory datatypeFactory;
-
-    private BundleContext context;
-
     private final TransformerManager schemaTransformerManager;
 
     private final TransformerManager mimeTypeTransformerManager;
 
-    private Validator validator;
+    private final TransformerManager inputTransformerManager;
+
+    private final ObjectFactory objectFactory = new ObjectFactory();
+
+    private final BundleContext context;
+
+    private final Validator validator;
+
+    private final CswQueryFactory queryFactory;
 
-    private CswQueryFactory queryFactory;
+    private final EventProcessor eventProcessor;
 
-    public CswSubscriptionEndpoint(BundleContext context,
+    private DatatypeFactory datatypeFactory;
+
+    private Map<String, ServiceRegistration<Subscription>> registeredSubscriptions =
+            new HashMap<>();
+
+    public CswSubscriptionEndpoint(BundleContext context, EventProcessor eventProcessor,
```

Comment:
Instead of injecting the bundle context, get it when needed with `FrameworkUtil`.